docs/eBPFWeakReferences.md

```diff
@@ -0,0 +1,82 @@
+
+**Introduction and problem statement**
+
+The eBPF user-kernel interface is built around a few key objects such as 'Programs', 'Maps' and their interaction. The 'eBPF for Windows' implementation makes extensive use of 'reference counting' to manage the interactions and lifetimes of these objects.  Code paths 'acquire' references to the required objects for the needed duration and 'release' them when done.  The reference counting design ensures that the referred-to objects remain valid (i.e. are not deleted) for the duration of the reference acquisition.
+
+By default, these references are of the 'strong' type, i.e. the referred-to objects are guaranteed to be valid for the duration of it's associated references.  While this property is mandatory in most use cases, it can lead to 'circular reference' instances.  Such instances are perfectly valid and the relationship between a 'Program Array Map' and a eBPF 'program' is an example of such a relationship.  This circular reference ensures that both the 'Program Array Map' remain valid for the duration of each other's existence, as they should be.
```

Comment:
<<< This circular reference ensures that both the 'Program Array Map' remain valid for the duration of each other's existence, as they should be.
>>> This circular reference ensures that both the 'Program Array Map' and the eBPF 'program' remain valid for the duration of each other's existence, as they should be.

Reply:
fixed